client/src/components/PlanningConflictForReport.js

```diff
@@ -0,0 +1,156 @@
+import { Icon, Intent, Spinner, Tooltip } from "@blueprintjs/core"
+import { IconNames } from "@blueprintjs/icons"
+import API from "api"
+import { gql } from "apollo-boost"
+import Report from "models/Report"
+import moment from "moment"
+import PropTypes from "prop-types"
+import React, { useEffect, useRef, useState } from "react"
+
+const GET_REPORT_WITH_ATTENDED_REPORTS = gql`
+  query($uuid: String, $attendedReportsQuery: ReportSearchQueryInput) {
+    report(uuid: $uuid) {
+      uuid
+      engagementDate
+      duration
+      attendees {
+        uuid
+        name
+        attendedReports(query: $attendedReportsQuery) {
+          list {
+            uuid
+            engagementDate
+            duration
+          }
+        }
+      }
+    }
+  }
+`
+
+const PlanningConflictForReport = ({ report, text, largeIcon }) => {
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState()
+  const [data, setData] = useState()
+
+  /*
+   * We cannot use API.useApiQuery here because this component is rendered by
+   * ReactDOM.render in ReportCalendar and ReportMap.
+   *
+   * If the component is unmounted before API.query is resolved or rejected,
+   * following warning is printed to the console;
+   *
+   * Warning: Can't perform a React state update on an unmounted component.
+   * This is a no-op, but it indicates a memory leak in your application.
+   * To fix, cancel all subscriptions and asynchronous tasks in a useEffect
+   * cleanup function.
+   *
+   * Since there is no way of canceling a Promise, _isMounted is a workaround.
+   *
+   * https://stackoverflow.com/questions/59780268/cleanup-memory-leaks-on-an-unmounted-component-in-react-hooks
+   */
```

Comment:
From that Apollo issue, the referenced blog post seems like a decent way to at least avoid the annoying warning: https://dev.to/rodw1995/cancel-your-promises-when-a-component-unmounts-gkl

Reply:
I don't even think we need useRef. This should work because of closure:
```
useEffect(() => {
  let canceled = false;
  ...
  if(!canceled) do_state_related_work();
  return () => {canceled = true};
}
```